frontend/src/app/core/services/auth.service.ts

```diff
@@ -41,11 +41,16 @@ export class AuthService {
     }
 
     signUp(email: string, password: string) {
-        return this.afAuth.createUserWithEmailAndPassword(email, password).then((userCredential) => {
-            if (userCredential.user) {
-                this.setAccessToken(userCredential.user);
-            }
-        });
+        return this.afAuth
+            .createUserWithEmailAndPassword(email, password)
+            .then(async (userCredential) => {
+                if (userCredential.user) {
+                    await this.setAccessToken(userCredential.user).then(() => this.navigateTo('topics'));
```

Comment:
it would be better to move this to a separate which will accept string value for defining a route

Reply:
and use it  method everywhere
as I can see we have at least two places with the same logic